eZ/Bundle/EzPublishCoreBundle/Command/NormalizeImagesPathsCommand.php

```diff
@@ -146,25 +179,25 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         return 0;
     }
 
-    private function updateImagePath(int $fieldId, string $oldPath, string $newPath, SymfonyStyle $io): void
-    {
+    /**
+     * @param resource $inputStream
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function updateImagePath(
+        int $fieldId,
+        string $oldPath,
+        string $newPath,
+        BinaryFile $oldBinaryFile,
+        $inputStream
+    ): void {
         $oldPathInfo = pathinfo($oldPath);
         $newPathInfo = pathinfo($newPath);
         // In Image's XML, basename does not contain a file extension, and the filename does - pathinfo results are exactly the opposite.
         $oldFileName = $oldPathInfo['basename'];
         $newFilename = $newPathInfo['basename'];
         $newBaseName = $newPathInfo['filename'];
 
-        // Checking if a file exists physically
-        $oldBinaryFile = $this->ioService->loadBinaryFileByUri(\DIRECTORY_SEPARATOR . $oldPath);
-        try {
-            $inputStream = $this->ioService->getFileInputStream($oldBinaryFile);
-        } catch (BinaryFileNotFoundException $e) {
-            $io->warning(sprintf('Skipping file %s as it doesn\'t exists physically.', $oldPath));
-
-            return;
-        }
-
         $xmlsData = $this->imageGateway->getAllVersionsImageXmlForFieldId($fieldId);
         foreach ($xmlsData as $xmlData) {
```

Comment:
What concerns me in the block below are the same calls at line 208 and 216
```php
$this->imageGateway->updateImagePath($fieldId, $oldPath, $newPath);
```
what is the purpose of performing the update twice? I don't see neither of the method arguments changing.

Reply:
As discussed on Slack resolved in https://github.com/ezsystems/ezpublish-kernel/pull/3139/commits/61a5472206e2fc06c9c59871b1bec1cd18d118c3